backend/src/controllers/rentals.py

```diff
@@ -109,6 +113,12 @@ def update(id):
                 if task.task_status != "completed":
                     task.task_status = "canceled"
         db.session.commit()
+    elif user_permissions == "client" and current_user.user_id == request.json["client_id"]:
```

Comment:
Czyli:
- sprawdzamy czy są uprawnienia klienta
- sprawdzamy czy id zalogowanego klienta zgadza się z przesłanym id
Jeżeli dobrze widzę, to mogę być zalogowana i podać swoje id robiąc request, ale anulować czyjąś rezerwację